static/js/utils/tile_utils.tsx

```diff
@@ -270,3 +276,88 @@ export function getSourcesJsx(sources: Set<string>): JSX.Element {
     </div>
   );
 }
+
+/**
+ * Gets the unit and scaling factor to use for a stat var spec
+ * @param svSpec stat var spec to get unit and scaling for
+ * @param statPointData stat data for the tile as a PointApiResponse
+ * @param statSeriesData stat data for the tile as a SeriesApiResponse
+ */
+export function getUnitAndScaling(
+  svSpec: StatVarSpec,
+  statPointData?: PointApiResponse,
+  statSeriesData?: SeriesApiResponse
+): { unit: string; scaling: number } {
+  // If the stat var spec specifies a unit, use that unit
+  const result = {
+    unit: svSpec.unit,
+    scaling: svSpec.scaling,
+  };
+  // Otherwise, try to get the unit from the stat data
+  if (!result.unit) {
+    let statMetadata = null;
+    if (statPointData) {
+      const obsWithFacet = Object.values(
+        statPointData.data[svSpec.statVar]
+      ).find((obs) => !!obs.facet);
+      statMetadata = statPointData.facets[obsWithFacet.facet];
+    }
+    if (statSeriesData) {
+      const seriesWithFacet = Object.values(
+        statSeriesData.data[svSpec.statVar]
+      ).find((series) => !!series.facet);
+      statMetadata = statSeriesData.facets[seriesWithFacet.facet];
+    }
+    result.unit = getUnit(statMetadata);
+  }
+  // If this is a per capita case and no unit has been found, use the default
+  // per capita unit and scaling
+  if (svSpec.denom && !result.unit) {
+    result.unit = DEFAULT_PC_UNIT;
+    result.scaling = DEFAULT_PC_SCALING;
+  }
+  return result;
+}
+
+interface DenomInfo {
+  value: number;
+  date: string;
+  source: string;
+}
+
+/**
+ * Gets information needed to calculate per capita for a single stat data point.
```

Comment:
nit: add a note here about date alignment and that null is returned when the matching value is 0

Reply:
done